### PR TITLE
Widget header

### PR DESCRIPTION
**Pull request by ArtsiomYeliseyenka**

### CODE REVIEW

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widget/widget.jsx [hunk 1]
```diff
@@ -1,52 +1,103 @@
...
+import { SpinningPreloader } from 'components/preloaders/spinningPreloader';
+import { WidgetHeader } from './widgetHeader';
 import styles from './widget.scss';
 
 const cx = classNames.bind(styles);
 
-export class Widget extends PureComponent {
+@connect(
+  (state, ownProps) => ({
+    url: URLS.widget(activeProjectSelector(state), ownProps.widgetId),
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardItemPage/widgetsGrid/widget/widget.jsx]: I suggest adding `widgetId` to propTypes to make this dependency clear

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx [hunk 2]
```diff
@@ -86,7 +87,10 @@ export class WidgetsGrid extends PureComponent {
     this.isResizing = false;
     this.onGridChange(newLayout);
   };
-
+  onDeleteWidget = (id) => {
+    // TODO implement when Pawels PR will be mergerd.
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]: It is merged now. You may implement in this PR or in the another one

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widget/widget.jsx (hunk 1 above)
Comment (ArtsiomYeliseyenka) [reply app/src/pages/inside/dashboardItemPage/widgetsGrid/widget/widget.jsx]: fixed

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/modals/deleteWidgetModal/deleteWidgetModal.jsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      onConfirm: () => {},
+    },
+  };
+
+  getWarningMessage = () => {
+    const { intl, data, userId, isAdmin, userProjectRole } = this.props;
+    if (data.widget.share && data.widget.owner === userId) {
+      return intl.formatMessage(messages.deleteOwnWidgetWarning);
+    }
+    if (data.widget.owner !== userId && isAdmin && userProjectRole === PROJECT_MANAGER) {
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardItemPage/widgetsGrid/modals/deleteWidgetModal/deleteWidgetModal.jsx]: Looks like here can be used special permission utility function

File: app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx [hunk 4]
```diff
@@ -67,43 +79,71 @@ export class WidgetsGrid extends Component {
...
-        widgetSize: [oldWidgets[index].widgetSize[0], h],
-      }));
-    } else {
-      newWidgets = newLayout.map(({ i, x, y, w, h }) => ({
-        widgetId: i,
-        widgetPosition: [x, y],
-        widgetSize: [w, h],
-      }));
-    }
+    Object.keys(oldWidgetPosition).forEach((prop) => {
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardItemPage/widgetsGrid/widgetsGrid.jsx]:
```
const itemChanged = Object.keys(oldWidgetPosition)
  .some((prop) => oldWidgetPosition[prop] !== newWidgetPosition[prop]);
```

**Pull request merged**